Sources/Controllers/Cloud/CloudTrashItemMenuViewController.swift

```diff
@@ -0,0 +1,152 @@
+
+    // MARK: - Initialize
+
+    init(with trashItem: TrashItem) {
+        self.trashItem = trashItem
+        super.init()
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    // MARK: - UIViewController
+
+    override func viewWillDisappear(_ animated: Bool) {
+        super.viewWillDisappear(animated)
+
+        if !isProcessSelected {
+            delegate?.resetSelectedIndexPath()
+        }
+    }
+
+    // MARK: - Base UI
+
+    override func getTitle() -> String {
+        localizedString("shared_string_file_info")
+    }
+
+    override func getLeftNavbarButtonTitle() -> String {
+        localizedString("shared_string_close")
+    }
+
+    override func isNavbarSeparatorVisible() -> Bool {
+        false
+    }
+
+    // MARK: - Table Data
+
+    override func generateData() {
+        let fileSection = tableData.createNewSection()
+        let fileRow = fileSection.createNewRow()
+        fileRow.key = "trashItem"
+        fileRow.cellType = OASimpleTableViewCell.getIdentifier()
+
+        let optionsSection = tableData.createNewSection()
+
+        if !trashItem.isLocalDeletion {
+            let restoreRow = optionsSection.createNewRow()
+            restoreRow.key = "restoreFromTrash"
+            restoreRow.cellType = OARightIconTableViewCell.getIdentifier()
+            restoreRow.title = localizedString("restore_from_trash")
+            restoreRow.secondaryIconName = "ic_custom_reset"
+            restoreRow.secondaryIconTintColor = UIColor.iconColorActive
+        }
+
+        let downloadRow = optionsSection.createNewRow()
+        downloadRow.key = "downloadToDevice"
+        downloadRow.cellType = OARightIconTableViewCell.getIdentifier()
+        downloadRow.title = localizedString("download_to_device")
+        downloadRow.secondaryIconName = "ic_custom_device_download"
+        downloadRow.secondaryIconTintColor = UIColor.iconColorActive
+
+        let deleteSection = tableData.createNewSection()
+        let deleteRow = deleteSection.createNewRow()
+        deleteRow.key = "deleteImmediately"
+        deleteRow.cellType = OARightIconTableViewCell.getIdentifier()
+        deleteRow.title = localizedString("shared_string_delete_immediately")
+        deleteRow.setObj(UIColor.textColorDisruptive, forKey: "titleColor")
+        deleteRow.secondaryIconName = "ic_custom_trash_outlined"
+        deleteRow.secondaryIconTintColor = UIColor.iconColorDisruptive
+    }
+
+    override func getRow(_ indexPath: IndexPath) -> UITableViewCell? {
+        let item = tableData.item(for: indexPath)
+        if item.cellType == OASimpleTableViewCell.getIdentifier() {
+            var cell = tableView.dequeueReusableCell(withIdentifier: OASimpleTableViewCell.getIdentifier()) as? OASimpleTableViewCell
+            if cell == nil {
+                let nib = Bundle.main.loadNibNamed(OASimpleTableViewCell.getIdentifier(), owner: self, options: nil)
```

Comment:
Do not use this code. It's a deprecated approach. Instead, use tableView.register() in all cases. Example: tableView.register(UINib(nibName: OASimpleTableViewCell.getIdentifier(), bundle: nil), forCellReuseIdentifier: OASimpleTableViewCell.getIdentifier())